### ISSUE TITLE
 Unable to find a matching sales channel for the request

### ISSUE THREAD

**Issue by Schrank** [PAYONE-GmbH/shopware-6#137]
Maybe this is already fixed, but just in case someone has the same problem...

We ran in an issue right after placing an order with Shopware 6.3.5.4 and Payone 2.3.1:

<img width="481" alt="image" src="https://user-images.githubusercontent.com/379680/135476149-f20bf648-a1f4-4714-b185-252abb0eabc0.png">

> Oops! An Error Occurred
> 
> The server returned a "500 Internal Server Error".
> 
> Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.

# What happend?
For some reason the redirect url of payone is wrong:

    https://example.com/payone/redirect?hash=ODQ1ODcxNDZiZjJiZTFjNDQ1NzI3Njc4ZjE4ZGMzNGFjYjA4OWQxNmRkZTkyZjEzMGJiNzZiNDJlMjQxYWI4ZA%3D%3D

but it should be 

    https://example.com/DE/payone/redirect?hash=ODQ1ODcxNDZiZjJiZTFjNDQ1NzI3Njc4ZjE4ZGMzNGFjYjA4OWQxNmRkZTkyZjEzMGJiNzZiNDJlMjQxYWI4ZA%3D%3D

This leads to an interesting bug, which ended in the above error - which is not a symfony error page and I was unable to find any log, neither php, apache, mysql, nothing. Took a while to find, hence this post, so someone might have a shorter way in the future.

# Solution:
Add the url to the sales_channel as well: https://example.com/ without DE at the end.

# TL;DR
Make sure the redirect url is correct, for some reason in our case one is chosen which is not attached to the sales_channel.

The 500 error can be reproduced by setting in `\Shopware\Storefront\Framework\Routing\RequestTransformer::transform` `$salesChannel` `null`

**Comment by matheusgontijo** [comment]
@Schrank, this persists on Shopware 6.6.1.1 when custom paths are added after the slash, like `/en` or `/de`...

**Comment by matheusgontijo** [comment]
@Schrank by the way, just created an issue on Shopware repository:

https://github.com/shopware/shopware/issues/3685
